src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -728,31 +728,21 @@ public void SetLocation(Geometry geometry)
         {
             TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
 
-            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint))
+            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint) && typeof(Autodesk.DesignScript.Geometry.Point).IsAssignableFrom(geometry.GetType()))
```

Comment:
@sharadkjaiswal can you explain it more please? I need to check first if my Element has a LocationPoint and if the supplied geometry is type of point. Once those are fine I can apply the geometry to the element.
In the other case both, argument and element.location must be curves. Then the location curve is being applied.